src/shared-components/Header.jsx

```diff
@@ -11,6 +11,22 @@ const renderHeading = (type, heading) => {
   if (type === 'sub') return <h4>{heading}</h4>
 }
 
+const renderHeadingWithSelector = (type, heading, selector) => {
```

Comment:
This seems pretty peculiar to me. I don't think we really need the warning here or the distinction between header/non-header with the selector. Also, I think the callsite can just be `renderHeadingWithSelector(props.type, heading, props.selector)` and we can do the check inside the Header component.